src/program/utils/ffprobe.py

```diff
@@ -68,24 +68,95 @@ def duration_in_mins(self) -> float:
         return round(self.duration / 60, 2)
 
 
+def _build_metadata_from_probe(probe_data: dict, display_name: str) -> MediaMetadata:
+    """Build a MediaMetadata object from ffprobe JSON output.
+
+    Args:
+        probe_data: Parsed ffprobe JSON
+        display_name: Filename to record in metadata
+    """
+    format_info = probe_data.get("format", {})
+
+    metadata_dict = {
+        "filename": display_name,
+        "file_size": int(format_info.get("size", 0)),
+        "duration": round(float(format_info.get("duration", 0)), 2),
+        "format": (
+            format_info.get("format_name", "unknown").split(",")
+            if format_info.get("format_name")
+            else []
+        ),
+        "bitrate": int(format_info.get("bit_rate", 0)),
+    }
```

Comment:
Why not just instantiate `MediaMetadata` here?

```suggestion
    metadata = MediaMetadata(
        filename=display_name,
        file_size=int(format_info.get("size", 0)),
        duration=round(float(format_info.get("duration", 0)), 2),
        format=(
            format_info.get("format_name", "unknown").split(",")
            if format_info.get("format_name")
            else []
        ),
        bitrate=int(format_info.get("bit_rate", 0)),
    )
```

edit: nvm another lift and shift lol